src/components/UpdateAppModal/BaseUpdateAppModal.js

```diff
@@ -1,44 +1,33 @@
-import React, {PureComponent} from 'react';
+import React, {useState} from 'react';
 import {propTypes, defaultProps} from './updateAppModalPropTypes';
 import ConfirmModal from '../ConfirmModal';
 import withLocalize from '../withLocalize';
 
-class BaseUpdateAppModal extends PureComponent {
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            isModalOpen: true,
-        };
-
-        this.submitAndClose = this.submitAndClose.bind(this);
-    }
+function BaseUpdateAppModal(props) {
+    const [isModalOpen, setIsModalOpen] = useState(true);
 
     /**
      * Execute the onSubmit callback and close the modal.
      */
-    submitAndClose() {
-        this.props.onSubmit(this.state.file);
-        this.setState({isModalOpen: false});
+    function submitAndClose() {
```

Comment:
I guess better to have an arrow function here (readability, consistency between functional components)

Reply:
<img width="495" alt="Screenshot 2023-08-06 at 7 00 21 AM" src="https://github.com/Expensify/App/assets/108292595/0d8bfa68-2431-4b53-9848-622141101d97">

Reply:
@situchan Linting won't complain if I use it this way `onConfirm={() => submitAndClose()}`
Would that be ok?

Also, what do you think about @pasyukevich's suggestion? I saw that for component creation the recommendation is to use the function keyword, but did not see anything about functions within the component. Should I use arrow functions, normal ones or it's down to personal preference?

Reply:
Arrow functions are preferred, but there are some exceptions like our case, hook (i.e. useCallback), etc